lua/apisix/plugins/redirect.lua

```diff
@@ -0,0 +1,121 @@
+local core = require("apisix.core")
+local tab_insert = table.insert
+local tab_concat = table.concat
+local re_gmatch = ngx.re.gmatch
+local ipairs = ipairs
+
+local lrucache = core.lrucache.new({
+    ttl = 300, count = 100
+})
+
+
+local schema = {
+    type = "object",
+    properties = {
+        ret_code = {type = "integer", minimum = 200, default = 302},
+        uri = {type = "string", minLength = 2},
+    },
+    required = {"uri"},
+}
+
+
+local plugin_name = "rewrite"
+
+local _M = {
+    version = 0.1,
+    priority = 900,
+    name = plugin_name,
+    schema = schema,
+}
+
+
+local function parse_uri(uri)
+
+    local reg = [[ (\\\$[0-9a-zA-Z_]+) | ]]         -- \$host
+                .. [[ \$\{([0-9a-zA-Z_]+)\} | ]]    -- ${host}
+                .. [[ \$([0-9a-zA-Z_]+) | ]]        -- $host
+                .. [[ (\$|[^$\\]+) ]]               -- $ or others
+    local iterator, err = re_gmatch(uri, reg, "jiox")
+    if not iterator then
+        return nil, err
+    end
+
+    local t = {}
+    while true do
+        local m, err = iterator()
+        if err then
+            return nil, err
+        end
+
+        if not m then
+            break
+        end
+
+        tab_insert(t, m)
+    end
+
+    return t
+end
+
+
+function _M.check_schema(conf)
+    local ok, err = core.schema.check(schema, conf)
+    if not ok then
+        return false, err
+    end
+
+    local uri_segs, err = parse_uri(conf.uri)
+    if not uri_segs then
+        return false, err
+    end
+    core.log.info(core.json.delay_encode(uri_segs))
+
+    return true
+end
+
+
+    local tmp = {}
+local function concat_new_uri(uri, ctx)
+    local pased_uri_segs, err = lrucache(uri, nil, parse_uri, uri)
+    if not pased_uri_segs then
+        return nil, err
+    end
+
+    core.table.clear(tmp)
+
+    for _, uri_segs in ipairs(pased_uri_segs) do
+        local pat1 = uri_segs[1]
```

Comment:
what's mean of `pat1` ... `pat4`? which are hard to understand.

Reply:
This is related to the regular content. It’s hard to take a good name here. Do you have any suggestions?

Reply:
I added some comments.